src/main/groovy/org/jetbrains/gradle/ext/IdeArtifacts.groovy

```diff
@@ -167,12 +168,29 @@ class ArchiveArtifact extends RecursiveArtifact {
       it.buildTo(temp)
     }
 
-    def customArchName = name
-    project.tasks.create(randomName, Zip) {
-      from temp
-      destinationDir destination
-      archiveName customArchName
-    }.execute()
+    createTestArchive(name, temp, destination)
+  }
+
+  def createTestArchive(String archiveName, File inputDir, File destinationDir) {
+
+    ZipOutputStream output = new ZipOutputStream(new FileOutputStream(new File(destinationDir, archiveName)))
```

Comment:
Should probably use `.withCloseable` to ensure that the output is always closed.

Reply:
fixed